frontend/src/js/main.js

```diff
@@ -0,0 +1,72 @@
+const SEARCH_THRESHHOLD = 1000;
+const API_URI = '';
+const DOMAIN_ENDPOINT = API_URI + '/domain/';
+const RESULT_DIV = 'div#results';
+const SEARCH_INPUT = 'input#search-input';
+
+let searchInputTime = Date.now();
+let shouldSearch = false;
+
+const searchDomain = async (domain) => {
+  const response = await fetch('http://localhost:3000' + DOMAIN_ENDPOINT + domain);
+
+  return response.json();
+};
+
+const processQuery = async (domain) => {
+  const resp = await searchDomain(domain);
+
+  console.log(resp);
+  const resultElement = document.querySelector(RESULT_DIV);
+  const searchElement = document.querySelector(SEARCH_INPUT);
+
+  for (let [tld, data] of Object.entries(resp)) {
+    if (Object.keys(data).indexOf('available') !== -1 && data['available'] === true) {
```

Comment:
Можно просто вот так. Если такого ключа нет, то `data.available === undefined`
```suggestion
    if (data.available === true) {
```